lib/evmone/eof.cpp

```diff
@@ -430,6 +431,27 @@ std::variant<EOFValidationError, int32_t> validate_max_stack_height(
     return max_stack_height;
 }
 
+std::set<size_t> update_accessed_code_sections(
+    std::set<size_t> accessed_code_sections,
+    const EOF1Header& header,
+    size_t code_index,
+    bytes_view container) noexcept
+{
+    const bytes_view code{header.get_code(container, code_index)};
+    for (size_t i = 0; i < code.size(); ++i)
+    {
+        const auto opcode = code[i];
+        if (opcode == OP_CALLF)
+        {
+          const auto fid = read_uint16_be(&code[i + 1]);
+          accessed_code_sections.insert(fid);
+          i += 2;
+        } else
+          i += instr::traits[opcode].immediate_size;
```

Comment:
should have a special case for `RJUMPV` like in https://github.com/ethereum/evmone/blob/master/lib/evmone/eof.cpp#L368

Reply:
Maybe better even to do it in `validate_instructions()` not  to repeat immediates handling. E.g. pass `accessed_code_sections` by reference there.

Reply:
Thanks for reviewing, I moved it to the `validate_instructions()`, it looks simpler.